### ISSUE TITLE
Unparseable section on CREATE DATABASE with DEFAULT CHARACTER SET

### ISSUE THREAD

**Issue by morags** [sqlfluff/sqlfluff#4259]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parser raises "Found unparsable section" error on the following:
```sql
CREATE DATABASE IF NOT EXISTS 'db_name' DEFAULT CHARACTER SET utf8;
```

The same thing happens with this example, taken from the [MariaDB documentation](https://mariadb.com/kb/en/create-database/):
```sql
CREATE DATABASE czech_slovak_names 
  CHARACTER SET = 'keybcs2'
  COLLATE = 'keybcs2_bin';
```

### Expected Behaviour

Parser should parse the section correctly.

### Observed Behaviour

See above.

### How to reproduce

1. Save the code given above in an `.sql` file
2. Specify the "mysql" dialect either in `.sqlfluff`, or as a commandline parameter
3. Run `sqlfluff lint filename.sql`

### Dialect

mysql

### Version

1.4.5

### Configuration

```ini
[sqlfluff]
dialect = mysql
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by yoichi** [comment]
The first example is not valid as 

```
mysql> CREATE DATABASE IF NOT EXISTS 'db_name' DEFAULT CHARACTER SET utf8;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''db_name' DEFAULT CHARACTER SET utf8' at line 1
```
tested with MySQL 8.0.29

Anyway, sqlfluff does not yet support the create_option documented at https://dev.mysql.com/doc/refman/8.0/en/create-database.html. I will issue a PR.
